devito/propagator.py

```diff
@@ -235,36 +234,183 @@ def prep_variable_map(self):
         var_map[self.time_dim] = symbols("i%d" % (i + 1))
         self._var_map = var_map
 
+    def factorise(self, stencils):
```

Comment:
Can this be separated into a standalone function to be applied over sympy expressions? In light of PR #92 we might want to decouple this functionality from the actual Propagator.